### PR TITLE
Rename `Core` to avoid collisions with other packages (i.e. GraphViz)

### PR DESCRIPTION
**Pull request by elliottwilliams**
Target names must be globally unique. GraphViz added a `Core` target in https://github.com/SwiftDocOrg/GraphViz/pull/15, which collides with XcodeGen's `Core` target (meaning you can cause a build failure with `swift package update && swift build`). I'm going to open a PR in GraphViz to rename their target as well, but we should really do this in both repos, to avoid situations like this in the future.

I tried to follow the convention based on "XcodeGenKit" and "XcodeGenCLI" but please feel free to bikeshed on the name 🙂 

[SE-0226](https://github.com/apple/swift-evolution/blob/master/proposals/0226-package-manager-target-based-dep-resolution.md), once it's fully implemented, avoids target name collisions for _inactivate_ targets, which will allow for multiple packages to have same-named test targets, for instance. But the "minimal set" of targets to build still need to be uniquely named. There are some pretty deep conceptual issues which prevent a package from depending on two different swift modules that have the same name: https://forums.swift.org/t/multiple-target-issue-with-spm/16696/4

### CODE REVIEW

**Comment by freddi-kit** [comment]
Hi, @elliottwilliams! Thank you for contribution! 
This is already done by other PR (https://github.com/yonaskolb/XcodeGen/pull/1049). Can you check it?

**Comment by elliottwilliams** [comment]
@freddi-kit oh thanks, I missed that PR!

Seems like it also solves the problem. The only difference is that instead of moving the source files to a new `XcodeGenCore` directory, it provides a non-default sources path: https://github.com/yonaskolb/XcodeGen/pull/1049/files#diff-f913940c58e8744a2af1c68b909bb6383e49007e6c5a12fb03104a9006ae677eL55-R55

IMO it's cleaner to just move the files, but I can see the argument for keeping them at their current paths.

**Comment by freddi-kit** [comment]
@yonaskolb Can you help to review it? this also helps to fix https://github.com/yonaskolb/XcodeGen/issues/1048. The difference with https://github.com/yonaskolb/XcodeGen/pull/1049 is directories name changing or not, please check which is prefer.

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 #### Internal
 - Build universal binaries for release. XcodeGen now runs natively on Apple Silicon. [#1024](https://github.com/yonaskolb/XcodeGen/pull/1024) @thii
+- The `Core` target is renamed to avoid collisions with other packages. [#1057](https://github.com/yonaskolb/XcodeGen/pull/1057) @elliottwilliams
```
Comment (freddi-kit) [inline CHANGELOG.md:22]: Please add it to `Next Version`. This field is for already released version
https://github.com/yonaskolb/XcodeGen/pull/1057/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edL7

Comment (yonaskolb) [reply CHANGELOG.md:22]: Will fix on master

**Review by yonaskolb — APPROVED** [review]
Thanks @elliottwilliams. Strange that we have to do this and SwiftPM doesn't handle it for us

**Pull request merged**